plugin/core/sessions.py

```diff
@@ -160,28 +176,78 @@ def __init__(self,
         self._on_post_exit = on_post_exit
         self.capabilities = dict()  # type: Dict[str, Any]
         self.client = client
+        self._workspace_folders = workspace_folders
         if on_pre_initialize:
             on_pre_initialize(self)
-        self._initialize(project_path)
+        self._initialize()
 
     def has_capability(self, capability: str) -> bool:
         return capability in self.capabilities and self.capabilities[capability] is not False
 
     def get_capability(self, capability: str) -> 'Optional[Any]':
         return self.capabilities.get(capability)
 
-    def _initialize(self, project_path: str) -> None:
-        params = get_initialize_params(project_path, self.config)
+    def handles_path(self, file_path: 'Optional[str]') -> bool:
```

Comment:
`Optional[str]` here because almost all the time we end up wanting to pass `view.file_name()` into it.